### DISCUSSION TITLE
Tried to install on AWS Bitnami LAMP Server (debian)

### DISCUSSION THREAD

**Discussion by inglesuniversal** [coder/code-server#3469] [Q&A] [answered]
All the steps worked with current version

but server will not install caddy 

and

port 8080 shows as being used

Any pointers or data you need to help me figure what is missing?

*****

$ code-server --open
[2021-05-24T20:39:27.625Z] info  code-server 3.10.2 387b12ef4ca404ffd39d84834e1f0776e9e3c005
[2021-05-24T20:39:27.627Z] info  Using user-data-dir ~/.local/share/code-server
[2021-05-24T20:39:27.634Z] error listen EADDRINUSE: address already in use 127.0.0.1:8080

**Comment by bpmct** [comment, marked as answer]
Converted this to a discussion.

Do you know what service is running on port 8080? Maybe code-server was added as a service and started automatically, removing the need to start it manually.

If not, I would say to simply launch code-server with a different port with

```sh
code-server --bind-addr 127.0.0.1:8081
```

and edit the Caddy config accordingly. Sadly, we can't help with installing Caddy on a server.

However, if you're using a LAMP server, I wouldn't recommend an additional Caddy server. Two options

1. Start `code-server --link` and use our tunnel. No need to reverse proxy or set up domains. This would also fix your port issue.
2. Use apache2 proxypass to proxy code-server. [Here is the unofficial config](https://github.com/cdr/code-server/issues/1544#issuecomment-743912980) but it hasn't been tested in a while. Make sure to change from 8080 to 8081 if you end up changing the port.

**Comment by inglesuniversal** [comment]
Wow... I tested the beta --link option and I was completely blown away!

It brought to my memory... the good old days of CODIAD but 2000 years ahead of time... Congrats to you and your TEAM.

I was so indecisive, between a dozen cloud IDE's including Cloud9 from AWS... But this one... really shows a fine degree of code craftsmanship which I really admire.

Well, let me get around the interface and I'll sure add more stuff and findings as I check the insights of your project.

#STAYSAFE The WORLD needs Developers Like you!
